src/components/modals/EditDishesModal.jsx

```diff
@@ -0,0 +1,180 @@
+import React, { Component } from "react";
+import {
+  Button,
+  Modal,
+} from "react-bootstrap";
+import { generateId } from "../../utils";
+
+class EditDishesModal extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      menuItem: this.props.showEditDishesModal.item,
+    };
+    
+    this.itemNameRef = React.createRef();
+    this.itemPriceRef = React.createRef();
+    this.itemQuantityRef = React.createRef();
+    this.ingredientsRef = React.createRef();
+  }
+
+  handlemenuItemChange = (event) => {
+    const { menuItem } = this.state;
+    const { name, value } = event.target;
+    if (name === 'name') {
+      this.itemNameRef.current.classList.remove('is-invalid');
+      this.itemNameRef.current.placeholder = '';
+      if(!menuItem.id) {
+        menuItem.id = generateId();
+      }
+      menuItem.name = value;
+    } else {
+      this.itemPriceRef.current.classList.remove('is-invalid');
+      this.itemPriceRef.current.placeholder = '';
+      this.itemQuantityRef.current.classList.remove('is-invalid');
+      this.itemQuantityRef.current.placeholder = '';
+      this.ingredientsRef.current.classList.remove('is-invalid');
+      this.ingredientsRef.current.placeholder = '';
+      menuItem[name] = value;
+    }
+    this.setState({ menuItem });
+  }
+
+  saveChanges = () => {
+    if(!this.validateInputFields()) {
+      return;
+    }
+
+    this.props.saveDetailsChange(this.state.menuItem);
+  }
+
+  closeModal = () => {
+    this.setState({menuItem: {
+      id: '',
+      name: '',
+      price: '',
+      ingredients: '',
+      quantity: '',
+    }}, () => {
+      this.props.closeEditDishesModal();
+    });
+  }
+
+  validateInputFields = () => {
+    let isValid = true;
+    if(!this.itemNameRef.current.value) {
+      this.itemNameRef.current.classList.add('is-invalid');
+      this.itemNameRef.current.placeholder = 'Meal name is required';
+      isValid = false;
+    }
+    if(!this.itemPriceRef.current.value) {
+      this.itemPriceRef.current.classList.add('is-invalid');
+      this.itemPriceRef.current.placeholder = 'Meal price is required';
+      isValid = false;
+    }
+    if(!this.itemQuantityRef.current.value) {
+      this.itemQuantityRef.current.classList.add('is-invalid');
+      this.itemQuantityRef.current.placeholder = 'Quantity of the meal is required';
+      isValid = false;
+    }
+    if(!this.ingredientsRef.current.value) {
+      this.ingredientsRef.current.classList.add('is-invalid');
+      this.ingredientsRef.current.placeholder = 'Please provide ingredients';
+      isValid = false;
+    }
+
+    return isValid;
+  }
+
+  render() {
+    const { menuItem } = this.state;
+
+    return (
+      <Modal
+        show={this.props.showEditDishesModal.show}
+        onShow={() => {
+          this.setState({ menuItem: {
```

Comment:
{id, name, price, etc } = this.props.showEditDishesModal.item they have the same key so values can be added like this